### PR TITLE
Piff cenc fixes

### PR DESCRIPTION
**Pull request by rbouqueau**

### CODE REVIEW

File: src/isomedia/drm_sample.c [hunk 1]
```diff
@@ -1181,13 +1181,14 @@ GF_Err gf_isom_cenc_get_sample_aux_info(GF_ISOFile *the_file, u32 trackNumber, u
 	}
 
 	gf_isom_get_sample_cenc_info_ex(trak, NULL, sampleNumber, NULL, &IV_size, NULL);
-	if (!IV_size) {
+	if (!IV_size && !type) {
+		/*not CENC and no encryption box: return void*/
```
Comment (nguyen-viet-thanh-trung) [inline src/isomedia/drm_sample.c]: should be initilized

File: src/isomedia/drm_sample.c [hunk 2]
```diff
@@ -1181,13 +1181,14 @@ GF_Err gf_isom_cenc_get_sample_aux_info(GF_ISOFile *the_file, u32 trackNumber, u
...
 		GF_SAFEALLOC( (*sai),  GF_CENCSampleAuxInfo);
 		return GF_OK;
 	}
 
 	/*get sample auxiliary information by saiz/saio rather than by parsing senc box*/
-	if (gf_isom_cenc_has_saiz_saio_track(stbl)) {
+	if (IV_size && gf_isom_cenc_has_saiz_saio_track(stbl)) {
```
Comment (nguyen-viet-thanh-trung) [inline src/isomedia/drm_sample.c]: should be limited to piff

File: src/isomedia/drm_sample.c [hunk 3]
```diff
@@ -1205,12 +1206,15 @@ GF_Err gf_isom_cenc_get_sample_aux_info(GF_ISOFile *the_file, u32 trackNumber, u
 	if (!a_sai)
 		return GF_NOT_SUPPORTED;
 
-	GF_SAFEALLOC( (*sai),  GF_CENCSampleAuxInfo);
+	GF_SAFEALLOC((*sai), GF_CENCSampleAuxInfo);
 	if (a_box) {
-		memmove((*sai)->IV, a_sai->IV, 16);
+		u8 size = ((*sai)->IV_size != 0) ? (*sai)->IV_size : 8/*default for modern PIFF/CENC with AES-CTR*/;
```
Comment (nguyen-viet-thanh-trung) [inline src/isomedia/drm_sample.c]: IV_size should be initilized

File: src/isomedia/track.c [hunk 4]
```diff
@@ -701,17 +700,21 @@ GF_Err MergeTrack(GF_TrackBox *trak, GF_TrackFragmentBox *traf, u64 moof_offset,
...
+					GF_CENCSampleAuxInfo *sai;
+					char *buffer;
+					GF_BitStream *bs;
+					u64 cur_position;
 					if (nb_saio != 1)
 						offset = (saio->version ? saio->offsets_large[i] : saio->offsets[i]) + moof_offset;
 					size = saiz->default_sample_info_size ? saiz->default_sample_info_size : saiz->sample_info_size[i];
 
-					/*cur_position = gf_bs_get_position(trak->moov->mov->movieFileMap->bs);
+					cur_position = gf_bs_get_position(trak->moov->mov->movieFileMap->bs);
```
Comment (nguyen-viet-thanh-trung) [inline src/isomedia/track.c]: if (gf_isom_cenc_has_saiz_saio_traf(traf)) 
 gf_isom_cenc_merge_saiz_saio(senc, trak->Media->information->sampleTable, offset, size);
else 
 //comment block

**Comment by rbouqueau** [comment]
I'm closing: replaced by #787

**Pull request closed without merging**